### PR TITLE
Closes #272 | Create dataset loader for SNLI

### LINKED ISSUE #272: Create dataset loader for SNLI Indo
Dataloader name: `snli-indo/snli-indo.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?snli-indo

|     Dataset| snli-indo |
|-------------|---|
| Description | The SNLI Indo dataset is derived from the SNLI corpus by translating each premise and hypothesis sentence from English to Indonesia via the Google Cloud Translation API.   Premise sentences are crawled image captions from Flickr, and hypothesis sentences are manually created through crowdsourcing. Five annotators are assigned per sentence pair to label the inference relationship as entailment (true), contradiction (false) or neutral (undetermined). |
| Subsets | - |
| Languages | ind |
| Tasks | Semantic Textual Similarity |
| License | Creative Commons Attribution 4.0 (cc-by-4.0) |
| Homepage | https://data.mendeley.com/datasets/k4tjhzs2gd/1 |
| HF URL | - |
| Paper URL | https://www.sciencedirect.com/science/article/pii/S2352340923010284 |

### PR DESCRIPTION
**Pull request by khelli07**
Closes [#272 ](https://github.com/SEACrowd/seacrowd-datahub/issues/272)

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/snli/snli.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

image

### CODE REVIEW

File: seacrowd/sea_datasets/snli/snli.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        "netral": "neutral",
+    }
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "premise": datasets.Value("string"),
+                    "hypothesis": datasets.Value("string"),
+                    "label": datasets.ClassLabel(names=self.labels),
```
Comment (holylovenia) [inline seacrowd/sea_datasets/snli/snli.py:97]: For the `source` schema, let's keep it as close as possible to the original data format by keeping the label as the original string, i.e., the value of the `in_to_eng_labelmap`.

File: seacrowd/sea_datasets/snli/snli.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    """SNLI Indo is derived from the SNLI corpus, where the premise and hypothesis sentences are translated directly from English to Indonesian using the Google Cloud Translation API. This dataset contains ~570k annotated sentence pairs."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="snli_source",
+            version=SOURCE_VERSION,
+            description="snli source schema",
```
Comment (holylovenia) [inline seacrowd/sea_datasets/snli/snli.py:70]: "snli" --> "SNLI"

File: seacrowd/sea_datasets/snli/snli.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            schema="source",
+            subset_id="snli",
+        ),
+        SEACrowdConfig(
+            name="snli_seacrowd_pairs",
+            version=SEACROWD_VERSION,
+            description="snli SEACrowd schema",
```
Comment (holylovenia) [inline seacrowd/sea_datasets/snli/snli.py:77]: "snli" --> "SNLI"

File: seacrowd/sea_datasets/snli/snli.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+_LANGUAGES = ["ind"]
+
+_LICENSE = Licenses.CC_BY_4_0.value
+
+_LOCAL = False
+
+_URLS = {
+    _DATASETNAME: {
+        "train": "https://data.mendeley.com/public-files/datasets/k4tjhzs2gd/files/ee45b2bb-e2ea-47b7-bec4-b6653c467d27/file_downloaded",
+        "val": "https://data.mendeley.com/public-files/datasets/k4tjhzs2gd/files/23aff85c-ff72-48b6-aba1-c1dd5dac216b/file_downloaded",
```
Comment (holylovenia) [inline seacrowd/sea_datasets/snli/snli.py:48]: The validation data URL is identical to the test data URL. Is this intentional and as per how the dataset should be used?

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @khelli07, thank you for working on this dataloader. I have some suggestions and questions regarding this dataloader. Please see below.

**Review by ryanignatius — APPROVED** [review]
reviewed and tested

File: seacrowd/sea_datasets/snli/snli.py (hunk 4 above)
Comment (khelli07) [reply seacrowd/sea_datasets/snli/snli.py:48]: Thanks for the notice. Revised.

File: seacrowd/sea_datasets/snli/snli.py (hunk 1 above)
Comment (khelli07) [reply seacrowd/sea_datasets/snli/snli.py:97]: To confirm, so the label should be in bahasa Indonesia, isnt it?

Comment (holylovenia) [reply seacrowd/sea_datasets/snli/snli.py:92]: Yes, but for accessibility, could we add this comment `# ["contradiction", "entailment", "neutral"` beside the line, @khelli07?

**Review by holylovenia — CHANGES_REQUESTED** [review]
One minor fix needed.

File: seacrowd/sea_datasets/snli/snli.py (hunk 1 above)
Comment (khelli07) [reply seacrowd/sea_datasets/snli/snli.py:92]: I put it beside `labels` since both schema uses Indonesian label

**Review by holylovenia — APPROVED** [review]
Looks good to me. Thanks @khelli07!! Merging now.

**Pull request merged**